### PR TITLE
[DOC] Documentation for builtin confusionMatrix function.

### PR DESCRIPTION
**Pull request by wizard-420**
please review if any changes are there to be done.

### CODE REVIEW

**Comment by wizard-420** [comment]
> please review if any changes are there to be done.

I have also uploaded the output in private doc.

**Comment by j143** [comment]
Hi @wizard-420 - Try to make sure that there are only commits related to `ConfusionMatrix`.

Tip: Google these keywords - `how to rebase from command line`, `git rebase -i`

File: dev/docs/builtins-reference.md [hunk 1]
```diff
@@ -318,7 +320,47 @@ slicefinder(X,W, y, k, paq, S);
...
+### Usage
+```r
+confusionMatrix(P,Y)
+```
+
+### Arguments
+
+| Name    | Type                   | Default  | Description |
+| :------ | :-------------         |  :---    | :---------- |
+| P       |      Matrix[Double]    |   ---    |vector of prediction |
```
Comment (j143) [inline dev/docs/builtins-reference.md:344]: Can this be made consistent? - i.e., after `|` one space only, and one space in the end. Tables syntax for other functions is good for reference.

File: dev/docs/builtins-reference.md [hunk 2]
```diff
@@ -318,7 +320,47 @@ slicefinder(X,W, y, k, paq, S);
...
 w = lm(X = X, y = y)
 ress = slicefinder(X = X,W = w, Y = y,  k = 5, paq = 1, S = 2);
 ```
+
+## `confusionMatrix`-Function
+
+A `confusionMatrix` is a technique for summarizing the performance of a classification algorithm.
+Calculating a confusion matrix can give you a better idea of what your classification model is getting right and what types of errors it is making.
+This confusionMatrix function accepts two matrices with one column each, these two matrices are vector for prediction and one-hot-encoded matrix respectively.
+Then it computes the max value of each vector and compare them, after whichit calculates and returns the sum of classifications and the average of each true class.
```
Comment (j143) [inline dev/docs/builtins-reference.md:333]: These lines are long, can they be curtailed a bit, to be readable. 
For example `331` line we can stop the  line at `what types` and `of errors it is making.` comes in the `332` line.

File: dev/docs/builtins-reference.md [hunk 3]
```diff
@@ -318,7 +320,47 @@ slicefinder(X,W, y, k, paq, S);
...
-y = X %*% rand(rows=ncol(X), 1)
+y = X %*% rand(rows = ncol(X), cols = 1)
 w = lm(X = X, y = y)
 ress = slicefinder(X = X,W = w, Y = y,  k = 5, paq = 1, S = 2);
 ```
+
+## `confusionMatrix`-Function
+
+A `confusionMatrix` is a technique for summarizing the performance of a classification algorithm.
+Calculating a confusion matrix can give you a better idea of what your classification model is getting right and what types of errors it is making.
```
Comment (j143) [inline dev/docs/builtins-reference.md:331]: `you`, `your` if possible can be avoided, instead. So, it is (without these two words!)
`a confusion matrix can give a better idea of what the classification model is getting right.`

File: dev/docs/builtins-reference.md [hunk 4]
```diff
@@ -318,7 +320,47 @@ slicefinder(X,W, y, k, paq, S);
...
+|:-----------------| :------------- | :---------- |
+|ConfusionSum      | Matrix[Double] | The Confusion Matrix Sums of classifications |
+|ConfusionAvg      | Matrix[Double] | The Confusion Matrix averages of each true class|
+
+### Example
+ #here numClasses is assigned to 1 as numClasses is directly proportional to the 
+ number of columns in the one hot data matrix, as confusion matrix accepts only matrices with one column.
+ 
+```r
+numClasses = 1
```
Comment (j143) [inline dev/docs/builtins-reference.md:359]: Can this be added in the example snippet itself.
```r
 # here numClasses is assigned to 1 as numClasses is directly proportional to the 
 # number of columns in the one hot data matrix, as confusion matrix accepts only matrices with one column.
```

File: dev/docs/builtins-reference.md [hunk 5]
```diff
@@ -318,7 +320,47 @@ slicefinder(X,W, y, k, paq, S);
...
+### Returns
+ 
+|Name  	          	| Type           | Description |
+|:-----------------| :------------- | :---------- |
+|ConfusionSum      | Matrix[Double] | The Confusion Matrix Sums of classifications |
+|ConfusionAvg      | Matrix[Double] | The Confusion Matrix averages of each true class|
+
+### Example
+ #here numClasses is assigned to 1 as numClasses is directly proportional to the 
+ number of columns in the one hot data matrix, as confusion matrix accepts only matrices with one column.
```
Comment (j143) [inline dev/docs/builtins-reference.md:356]: These lines can be removed.

File: scripts/builtin/outlier.dml [hunk 6]
```diff
@@ -18,6 +18,13 @@
 # under the License.
 #
 #-------------------------------------------------------------
+#An outlier in a probability distribution function is a number that is more 
+#than 1.5 times the length of the data set away from either the lower or upper quartiles.
+#Specifically, if a number is less than Q1−1.5×IQR or greater than Q3+1.5×IQR, then it is an outlier.
+#
+
+
+
```
Comment (j143) [inline scripts/builtin/outlier.dml:27]: Shall we delete thi

File: dev/docs/builtins-reference.md (hunk 5 above)
Comment (wizard-420) [reply dev/docs/builtins-reference.md:356]: ok I will make all the neccecary changes

File: scripts/builtin/outlier.dml (hunk 6 above)
Comment (wizard-420) [reply scripts/builtin/outlier.dml]: changes have been applied

**Comment by j143** [comment]
Thank you, @wizard-420 - LGTM. 👍 
🎉 🚀

**Pull request closed without merging**